src/types.tsx

```diff
@@ -62,6 +62,10 @@ export interface FormikComputedProps<Values> {
   readonly initialValues: Values;
 }
 
+/* Allow either raw values, or a method that takes values and returns values */
+export type ValueFunction<Values> = (values: Values) => Values;
+export type ValueUpdate<Values> = Values | ValueFunction<Values>;
+
```

Comment:
Not sure about type names

Reply:
I recommend: `ValuesCallback`, `ValuesOrCallback` (like you suggested above)

Reply:
gosh, naming is hard. How about this?

```ts
interface ValuesSetter<V> {
    (values: Values): void;
    (valuesFactory: (prevValues: Readonly<V>) => V): void;
}

// and then
{
    setValues: ValuesSetter<Values>
}
```

Reply:
I like `ValuesFactory` and use of `Readonly`. I think describing the expected type (`ValuesOrValuesFactory = Values | ValuesFactory`) or describing the Type's usage (`SetValuesParam`) works best. `ValuesSetter` makes me think more of the actual setValues function, if it were to have a type.

Reply:
Agree